src/perftest_parameters.c

```diff
@@ -114,6 +114,47 @@ static int parse_ethertype_from_str(char *ether_str, uint16_t *ethertype_val)
 	return SUCCESS;
 }
 
+static int parse_flow_label_from_str(struct perftest_parameters *user_param, char *flow_label_str)
+{
+	int fl_cnt = 1;
+	const char* sep = NULL;
+
+	sep = strchr(flow_label_str, ',');
+	if (sep != NULL)
+		do {
+			fl_cnt++;
+			sep = strchr(sep + 1, ',');
+		} while (sep);
+
+	int *flow_label = calloc(fl_cnt + 2, sizeof(int));
+	flow_label[0] = fl_cnt;
+	flow_label[1] = 0;
+	flow_label[2] = strtol(flow_label_str, NULL, 0);
+	sep = strchr(flow_label_str, ',');
+
+	for (int i = 3; i < fl_cnt + 2; i++) {
+		flow_label[i] = strtol(sep + 1, NULL, 0);
+		sep = strchr(sep + 1, ',');
+	}
+
+	if (user_param->connection_type == RawEth) {
+		for (int i = 2; i < fl_cnt + 2; i++) {
+			if (flow_label[i] < 0) {
+				fprintf(stderr," flow label must be non-negative for RawEth\n");
+				return -1;
+			}
+		}
+	}
+
+	if (user_param->flow_label) {
```

Comment:
in which scenario this will be true?

Reply:
@sshaulnv Good catch, it seems that's dead code. I'll remove this if-condition and feedback later.

Reply:
@sshaulnv This scenario is to deal with the case that user config the command option ```--flow_label=xxxx``` for more than one time. This is rare case.